File: src/main/java/cz/cvut/kbss/changetracking/exception/ClassNotAuditedException.java
package cz.cvut.kbss.changetracking.exception;

/**
 * {@link ClassNotAuditedException} is used when trying to run {@link cz.cvut.kbss.changetracking.ChangeTracker#compare}
 * on an object whose class is not supported for change tracking. With the default {@link
 * cz.cvut.kbss.changetracking.strategy.entity.EntityStrategy}, this means the class is missing the {@link
 * cz.cvut.kbss.changetracking.annotation.Audited} annotation).
 */
public class ClassNotAuditedException extends ChangeTrackingException {
	public ClassNotAuditedException(Class<?> clazz) {
		super("Object class doesn't support auditing: " + clazz.getCanonicalName());
	}
}

File: src/main/java/cz/cvut/kbss/changetracking/exception/ClassNotAuditedException.kt
package cz.cvut.kbss.changetracking.exception

/**
 * [ClassNotAuditedException] is used when trying to run [cz.cvut.kbss.changetracking.ChangeTracker.compare]
 * on an object whose class is not supported for change tracking. With the default
 * [cz.cvut.kbss.changetracking.strategy.entity.EntityStrategy], this means the class is missing the
 * [cz.cvut.kbss.changetracking.annotation.Audited] annotation.
 */
class ClassNotAuditedException(clazz: Class<*>) :
	ChangeTrackingException("Object class doesn't support auditing: " + clazz.name)
